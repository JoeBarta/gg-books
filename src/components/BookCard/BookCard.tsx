import React from "react"
import {
  Card,
  CardContent,
  CardHeader,
  makeStyles,
  Typography,
  Icon,
} from "@material-ui/core"
import LibraryBooksIcon from '@material-ui/icons/LibraryBooks'

import { Book } from "../../api/booksApi"

const useStyles = makeStyles((theme) => ({
  root: {
    backgroundColor: theme.palette.secondary.main,
    maxWidth: 275,
    color: theme.palette.primary.main,
  },
  content: {
    paddingTop: 0,
    paddingLeft: 16,
    paddingRight: 16,
    color: theme.palette.primary.main,
  },
}))

const BookCard = ({
  book_author,
  book_publication_city,
  book_publication_country,
  book_publication_year,
  book_pages,
  book_title
}: Book) => {
  const classes = useStyles()
  return (
    <Card className={classes.root}>
      <CardHeader
        className={classes.root}
        title={`Title: ${book_title}`}
        subheader= {`By: ${book_author[0]}`}
      />
      {book_author.length > 1 && <Typography variant='subtitle2'>Also Written by: </Typography>}
      {book_author.length > 1
        ? book_author.map((author: string, idx: number) => {
          if (idx !== 1) {
            return <Typography variant="body2" color="textSecondary" component="p">{author}</Typography>
          }
          return
        })
        : null
      }
      <CardContent className={classes.content}>
        <Icon>
          <LibraryBooksIcon />
        </Icon>
        <Typography variant='subtitle2'>Page Count: {book_pages}</Typography>
        <Typography variant="body2" color="textPrimary" component="p">Publishing details</Typography>
        <Typography variant="body2" color="textSecondary" component="p">City: {book_publication_city}</Typography>
        <Typography variant="body2" color="textSecondary" component="p">Country: {book_publication_country}</Typography>
        <Typography variant="body2" color="textSecondary" component="p">Year: {book_publication_year}</Typography>
      </CardContent>
    </Card>
  )
}

export default BookCard
