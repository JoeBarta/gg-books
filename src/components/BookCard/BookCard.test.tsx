import React from 'react'
import BookCard from './BookCard'
import { render } from '@testing-library/react'

const props = {
  book_author: ['George Michael'],
  book_publication_city: 'Landan mate',
  book_publication_country: 'UK',
  book_publication_year: '2003',
  book_pages: 420,
  book_title: 'Snappy Snaps, my memoir',
  id: 1,
}

test('renders as expected', () => {
  const { asFragment, getByText } = render(<BookCard {...props} />)
  expect(asFragment()).toMatchSnapshot()

  expect(getByText(`Title: ${props.book_title}`)).toBeInTheDocument()
  expect(getByText(`By: ${props.book_author[0]}`)).toBeInTheDocument()
  expect(getByText(`City: ${props.book_publication_city}`)).toBeInTheDocument()
  expect(getByText(`Country: ${props.book_publication_country}`)).toBeInTheDocument()
  expect(getByText(`Year: ${props.book_publication_year}`)).toBeInTheDocument()
  expect(getByText('Publishing details')).toBeInTheDocument()
})
