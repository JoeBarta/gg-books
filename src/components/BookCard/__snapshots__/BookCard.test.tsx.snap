// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardHeader-root makeStyles-root-1"
    >
      <div
        class="MuiCardHeader-content"
      >
        <span
          class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
        >
          Title: Snappy Snaps, my memoir
        </span>
        <span
          class="MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
        >
          By: George Michael
        </span>
      </div>
    </div>
    <div
      class="MuiCardContent-root makeStyles-content-2"
    >
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M4 6H2v14c0 1.1.9 2 2 2h14v-2H4V6zm16-4H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-1 9H9V9h10v2zm-4 4H9v-2h6v2zm4-8H9V5h10v2z"
          />
        </svg>
      </span>
      <h6
        class="MuiTypography-root MuiTypography-subtitle2"
      >
        Page Count: 420
      </h6>
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextPrimary"
      >
        Publishing details
      </p>
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        City: Landan mate
      </p>
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        Country: UK
      </p>
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        Year: 2003
      </p>
    </div>
  </div>
</DocumentFragment>
`;
